tests/ops/op_math/test_sprod.py

```diff
@@ -0,0 +1,635 @@
+        sprod_op = SProd(scalar, op)
+        assert op_rep == sprod_op.__repr__()
+
+    @pytest.mark.parametrize("op_scalar_tup", ops)
+    def test_copy(self, op_scalar_tup):
+        """Test the copy dunder method properly copies the operator."""
+        scalar, op = op_scalar_tup
+        sprod_op = SProd(scalar, op)
+        copied_op = copy(sprod_op)
+
+        assert sprod_op.scalar == copied_op.scalar
+
+        assert sprod_op.id == copied_op.id
+        assert sprod_op.data == copied_op.data
+        assert sprod_op.wires == copied_op.wires
+
+        assert sprod_op.base.name == copied_op.base.name
+        assert sprod_op.base.wires == copied_op.base.wires
+        assert sprod_op.base.data == copied_op.base.data
+
+
+class TestMatrix:
+    @pytest.mark.parametrize("scalar", scalars)
+    @pytest.mark.parametrize("op_and_mat", non_param_ops)
+    def test_non_parametric_ops(self, scalar, op_and_mat):
+        """Test matrix method for a scalar product of non_parametric ops"""
+        op, mat = op_and_mat
+        sprod_op = SProd(scalar, op(wires=range(op.num_wires)))
+
+        sprod_mat = sprod_op.matrix()
+        true_mat = scalar * mat
+        assert np.allclose(sprod_mat, true_mat)
+
+    @pytest.mark.parametrize("scalar", scalars)
+    @pytest.mark.parametrize("op_and_mat", param_ops)
+    def test_parametric_ops(self, scalar, op_and_mat):
+        """Test matrix method for a scalar product of parametric ops"""
+        op, mat = op_and_mat
+        params = range(op.num_params)
+        sprod_op = SProd(scalar, op(*params, wires=range(op.num_wires)))
+
+        sprod_mat = sprod_op.matrix()
+        true_mat = scalar * mat(*params)
+        assert np.allclose(sprod_mat, true_mat)
+
+    @pytest.mark.parametrize("op", no_mat_ops)
+    def test_error_no_mat(self, op):
+        """Test that an error is raised if the operator doesn't
+        have its matrix method defined."""
+        sprod_op = SProd(1.23, op(wires=0))
+        with pytest.raises(MatrixUndefinedError):
+            sprod_op.matrix()
+
+    def test_sprod_ops_wire_order(self):
+        """Test correct matrix is returned when the wire_order arg is provided."""
+        scalar = 1.23
+        sprod_op = SProd(scalar, qml.Toffoli(wires=[2, 0, 1]))
+        wire_order = [0, 1, 2]
+        mat = sprod_op.matrix(wire_order=wire_order)
+
+        ccnot = math.array(
+            [
+                [1, 0, 0, 0, 0, 0, 0, 0],
+                [0, 1, 0, 0, 0, 0, 0, 0],
+                [0, 0, 1, 0, 0, 0, 0, 0],
+                [0, 0, 0, 1, 0, 0, 0, 0],
+                [0, 0, 0, 0, 1, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 0, 1],
+                [0, 0, 0, 0, 0, 0, 1, 0],
+                [0, 0, 0, 0, 0, 1, 0, 0],
+            ]
+        )
+
+        true_mat = scalar * ccnot
+        assert np.allclose(mat, true_mat)
+
+    templates_and_mats = (
+        (qml.QFT(wires=[0, 1, 2]), get_qft_mat(3)),
+        (
+            qml.GroverOperator(wires=[0, 1, 2]),
```

Comment:
The grover operator also provides its own matrix.

Reply:
Got it!